### PR TITLE
[FIX] gcc12: owning_view and viewable_range

### PR DESCRIPTION
**Pull request by eseiler**
New viewable_range concept 
owning_view, which is quite similar to the persist_view :)

See:
* https://github.com/gcc-mirror/gcc/commit/5e1b17f038671df1a6580ece4cd58ef91cc1e234
* http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2415r2.html

### CODE REVIEW

File: include/seqan3/utility/views/enforce_random_access.hpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ class view_enforce_random_access : public std::ranges::view_interface<view_enfor
               std::ranges::viewable_range<viewable_rng_t> &&
               std::constructible_from<urng_t, std::ranges::ref_view<std::remove_reference_t<viewable_rng_t>>>
     //!\endcond
-    explicit constexpr view_enforce_random_access(viewable_rng_t range) :
+    explicit constexpr view_enforce_random_access(viewable_rng_t && range) :
         view_enforce_random_access{std::views::all(range)}
```
Comment (SGSSGene) [inline include/seqan3/utility/views/enforce_random_access.hpp:77]: könnte hier auch ein std::move hin?
```suggestion
        view_enforce_random_access{std::views::all(range)}
```

File: test/unit/utility/views/repeat_n_test.cpp [hunk 2]
```diff
@@ -77,7 +77,8 @@ TEST(view, factory)
 
     // view
     {
-        auto view = std::string{"foobar"} | seqan3::detail::persist | std::views::take(3);
+        std::string str{"foobar"};
+        auto view = str | std::views::take(3);
```
Comment (SGSSGene) [inline test/unit/utility/views/repeat_n_test.cpp:81]: so viel schöner :-)

**Review by SGSSGene — CHANGES_REQUESTED** [review]

File: include/seqan3/std/ranges [hunk 3]
```diff
@@ -67,6 +67,11 @@ inline constexpr bool enable_borrowed_range<::std::ranges::iota_view<W, Bound>>
 template <class T>
 inline constexpr bool enable_borrowed_range<::std::ranges::ref_view<T>> = true;
 
+#if __cpp_lib_ranges >= 202110L
+template <class T>
+inline constexpr bool enable_borrowed_range<::std::ranges::owning_view<T>> = enable_borrowed_range<T>;
+#endif
+
```
Comment (SGSSGene) [inline include/seqan3/std/ranges:74]: Doesn't line 67-68 have to be in the 'else' clause?

**Review by SGSSGene — APPROVED** [review]

File: include/seqan3/utility/views/enforce_random_access.hpp (hunk 1 above)
Comment (eseiler) [reply include/seqan3/utility/views/enforce_random_access.hpp]: Could, but apparently doesn't have to, I added a `std::forward`

**Review by smehringer — APPROVED** [review]
LGTM

**Pull request merged**